Site wide improvements (#43)
wixplosives/stylable.io: docs/references/runtime.md

title: Runtime
---

> Imported Stylable stylesheets in JS contain minimal runtime code to help define the structure and state of the component.

Author: ```suggestion
Imported Stylable stylesheets in JavaScript contain minimal runtime code to help define the structure and state of the component.
```

<!-- prettier-ignore-start -->
```js
/* index.jsx - stylesheet runtime api */

> import {

Author: double space before the "from"

## Manual mapping

Any namespaced symbols, defined in the stylesheet, are exposed on maps from their local name to their target namespaced name:

<!-- prettier-ignore-start -->
```js
import { classes, vars, keyframes, layers } from 'style.st.css';

> // class mapping

Author: ```suggestion
// classes
```

classes.root;            // "style__root"
classes.label;           // "style__label"
classes.icon;            // "style__icon"
classes['dashed-part'];  // "style__dashed-part"

> // custom property mapping

Author: ```suggestion
// custom property
```

vars.color1              // "--style-color1"

// keyframes
keyframes.slide          // "style__slide"

> // keyframes

Author: ```suggestion
// layers
```

layers.theme             // "style__theme"
```
<!-- prettier-ignore-end -->

:::note

> The [root class](../references/root.md) is available even when it is not defined in the stylesheet.

Author: ```suggestion
The [root class](../references/root.md) is available even when it is not explicitly defined in the stylesheet.
```

:::

## Build vars

Local stylesheet [build vars](./variables.md) are exported with their set value.
<!-- prettier-ignore-start -->
```css
:vars {
    bg: black;
    color: gold;
}
```

```js
import { stVars } from 'style.st.css';

stVars.bg;    // "black"
stVars.color; // "gold"
```
<!-- prettier-ignore-end -->

## `st()` function

> A utility function for concatenating [classes](./class-selectors.md) and [custom states](./pseudo-classes.md).

Author: ```suggestion
A utility function for concatenating [classes](./class-selectors.md) and activating [custom states](./pseudo-classes.md).
```

> - the **second argument** can optionally receive state activation

Author: ```suggestion
- the optional **second argument** can receive either a state activation or another class to concatenate
```
